messaging/kafka-streams-reactive-messaging/src/main/java/io/quarkus/ts/messaging/kafka/reactive/streams/shutdown/ReadinessObserver.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkus.ts.messaging.kafka.reactive.streams.shutdown;
+
+import jakarta.enterprise.event.Observes;
+
+import io.quarkus.logging.Log;
+import io.quarkus.runtime.StartupEvent;
+import io.smallrye.reactive.messaging.health.HealthReport;
+import io.smallrye.reactive.messaging.providers.extension.HealthCenter;
+
+public class ReadinessObserver {
+
+    private static final int SLEEP_PERIOD = 250;
+
+    void observe(@Observes StartupEvent ignored, HealthCenter healthCenter) throws InterruptedException {
+        // Quarkus doesn't wait till channels are ready, but we don't want to start working till we can use Kafka
```

Comment:
Two nitpicks: 
1. This is a description of the whole method, how about moving it before the method?
2. Maybe it's just me, but this sentence is a bit overcomplicated and can be simplified by dropping negatives. E.g. "Quarkus application starts before channels are ready, but our tests require the whole Kafka pipeline to be fully operational" (I believe  you can come with something much better, but this is a general idea)

Reply:
1. ok 2. ok